lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -838,36 +838,67 @@ class ADContext {
   SILPassManager &getPassManager() const { return passManager; }
   Lowering::TypeConverter &getTypeConverter() { return module.Types; }
 
-  SmallVectorImpl<DifferentiableFunctionInst *> &
-  getDifferentiableFunctionInsts() {
-    return differentiableFunctionInsts;
+  /// Returns true if there are `DifferentiableFunctionInst` to process.
+  bool hasDifferentiableFunctionInsts() const {
+    return !differentiableFunctionInsts.empty();
   }
 
-  SmallPtrSetImpl<DifferentiableFunctionInst *> &
-  getProcessedDifferentiableFunctionInsts() {
-    return processedDifferentiableFunctionInsts;
+  /// Pick and remove a `DifferentiableFunctionInst` for processing.
+  /// Returns nullptr if none are available for processing.
+  DifferentiableFunctionInst* pickAndRemoveDifferentiableFunctionInst() {
```

Comment:
How about this more descriptive name?
```suggestion
  DifferentiableFunctionInst* popDifferentiableFunctionInstFromWorklist() {
```
"pop" is a shorter term-of-art than "pick and remove". "from worklist" clarifies where the `DifferentiableFunctionInst` came from.